src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -151,38 +119,20 @@ public ApiClient load( Class<?> key ) throws Exception {
-        deviceESID = entitySetsApi.getEntitySetId( DEVICES_ENTITY_SET_NAME );
-        dataESID = entitySetsApi.getEntitySetId( DATA_ENTITY_SET_NAME );
-        preProcessedESID = entitySetsApi.getEntitySetId( PREPROCESSED_DATA_ENTITY_SET_NAME );
-        recordedByESID = entitySetsApi.getEntitySetId( RECORDED_BY_ENTITY_SET_NAME );
-        usedByESID = entitySetsApi.getEntitySetId( USED_BY_ENTITY_SET_NAME );
-        participatedInESID = entitySetsApi.getEntitySetId( PARTICIPATED_IN_AESN );
-        userAppsESID = entitySetsApi.getEntitySetId( CHRONICLE_USER_APPS );
-        questionnaireESID = entitySetsApi.getEntitySetId( QUESTIONNAIRE_ENTITY_SET_NAME );
-        questionsESID = entitySetsApi.getEntitySetId( QUESTIONS_ENTITY_SET_NAME );
-        answersESID = entitySetsApi.getEntitySetId( ANSWERS_ENTITY_SET_NAME );
-        addressesESID = entitySetsApi.getEntitySetId( ADDRESSES_ENTITY_SET_NAME );
-        respondsWithESID = entitySetsApi.getEntitySetId( RESPONDS_WITH_ENTITY_SET_NAME );
-        notificationsESID = entitySetsApi.getEntitySetId( NOTIFICATION_ENTITY_SET_NAME );
-        partOfESID = entitySetsApi.getEntitySetId( PART_OF_ENTITY_SET_NAME );
-
-        durationPTID = edmApi.getPropertyTypeId( DURATION.getNamespace(), DURATION.getName() );
-        stringIdPTID = edmApi.getPropertyTypeId( STRING_ID_FQN.getNamespace(), STRING_ID_FQN.getName() );
-        personIdPSID = edmApi.getPropertyTypeId( PERSON_ID_FQN.getNamespace(), PERSON_ID_FQN.getName() );
-        dateLoggedPTID = edmApi.getPropertyTypeId( DATE_LOGGED_FQN.getNamespace(), DATE_LOGGED_FQN.getName() );
-        versionPTID = edmApi.getPropertyTypeId( VERSION_FQN.getNamespace(), VERSION_FQN.getName() );
-        modelPTID = edmApi.getPropertyTypeId( MODEL_FQN.getNamespace(), MODEL_FQN.getName() );
-        dateTimePTID = edmApi.getPropertyTypeId( DATE_TIME_FQN.getNamespace(), DATE_TIME_FQN.getName() );
-        titlePTID = edmApi.getPropertyTypeId( TITLE_FQN.getNamespace(), TITLE_FQN.getName() );
-        fullNamePTID = edmApi.getPropertyTypeId( FULL_NAME_FQN.getNamespace(), FULL_NAME_FQN.getName() );
-        recordTypePTID = edmApi.getPropertyTypeId( RECORD_TYPE_FQN.getNamespace(), RECORD_TYPE_FQN.getName() );
-        startDateTimePTID = edmApi.getPropertyTypeId( START_DATE_TIME.getNamespace(), START_DATE_TIME.getName() );
-        dataKey = edmApi.getEntityType( entitySetsApi.getEntitySet( dataESID ).getEntityTypeId() ).getKey();
+        // get entity setId map
+        entitySetIdMap = entitySetsApi.getEntitySetIds( ENTITY_SET_NAMES );
+
+        // get propertyTypeId map
+        Iterable<PropertyType> propertyTypes = edmApi.getPropertyTypes();
+        propertyTypeIdMap = StreamSupport
+                .stream( propertyTypes.spliterator(), false )
+                .collect( Collectors.toMap(
+                        propertyType -> propertyType.getType(),
+                        propertyType -> propertyType.getId()
+                ) );
```

Comment:
Use `ImmutableMap`. Also, we should also have another map to remove duplicate logic from `getParticipantDataHelper()`:

```
private final ImmutableMap<UUID, PropertyType> propertyTypesById;
private final ImmutableMap<FullQualifiedName, UUID> propertyTypeIdsByFQN;

propertyTypesById = StreamSupport
    .stream( edmApi.getPropertyTypes().spliterator(), false )
    .collect( ImmutableMap.toImmutableMap( PropertyType::getId, Function.identity() ) );

propertyTypeIdsByFQN = propertyTypesById
    .values()
    .stream()
    .collect( ImmutableMap.toImmutableMap( PropertyType::getType, PropertyType::getId ) );
```